python/ray/data/grouped_data.py

```diff
@@ -32,7 +35,6 @@ def __repr__(self) -> str:
         return "T" + self.data.__repr__()
 
 
-@PublicAPI
```

Comment:
@bveeramani: i remove this annotation at the class level; and if this is missing, we assume its constructor is private

So we omit the constructor in the doc now, let me know if you like it, thankks

<img width="1229" alt="Screenshot 2024-08-06 at 12 23 31 PM" src="https://github.com/user-attachments/assets/3d75cf1c-3ef1-4a05-a71a-1f2b30278a51">

Reply:
Nice